internal/cli/roles_permissions.go

```diff
@@ -70,22 +81,38 @@ func listRolePermissionsCmd(cli *cli) *cobra.Command {
 				inputs.ID = args[0]
 			}
 
-			var list *management.PermissionList
+			list, err := getWithPagination(
+				cmd.Context(),
+				inputs.Number,
+				func(opts ...management.RequestOption) (result []interface{}, hasNext bool, err error) {
+					permissionsList, err := cli.api.Role.Permissions(inputs.ID, opts...)
+					if err != nil {
+						return nil, false, err
+					}
+
+					for _, role := range permissionsList.Permissions {
+						result = append(result, role)
+					}
+					return result, permissionsList.HasNext(), nil
+				},
+			)
 
-			if err := ansi.Waiting(func() error {
-				var err error
-				list, err = cli.api.Role.Permissions(inputs.ID)
-				return err
-			}); err != nil {
-				return fmt.Errorf("An unexpected error occurred: %w", err)
+			if err != nil {
+				return fmt.Errorf("Failed to get permissions: %w", err)
```

Comment:
I'd suggest including the role id as well, e.g. "Failed to get permissions for role X".

Reply:
Done in 3b09f4c66cbaa6dd1787bee70c0b1d4e69a71f2a